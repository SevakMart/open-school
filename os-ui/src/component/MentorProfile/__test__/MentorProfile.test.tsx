import { render, screen } from '@testing-library/react';
import MentorCard from '../MentorProfile';

/* const MentorProfile = {
  name: 'John',
  surname: 'Smith',
  professionName: 'JS Developer',
  companyName: 'Google',
  courseCount: 5,
  userImgPath: 'https://image.shutterstock.com/image-vector/businessman-avatar-profile-picture-260nw-221565274.jpg',
  emailPath: 'fakeEmail.com',
  linkedinPath: 'https://linkedin.com/feed',
};

describe('Make unit tests on Mentor Profile', () => {
  test('Make a snapshot test', () => {
    const { asFragment } = render(<MentorCard mentor={{ ...MentorProfile }} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Verify if Mentor info is contained in the component', () => {
    render(<MentorCard mentor={{ ...MentorProfile }} />);
    const fullname = screen.queryByTestId('John Smith');
    const professionNameElem = screen.queryByTestId('JS Developer');
    const companyNameElem = screen.queryByTestId('Google');
    const courseCountElem = screen.queryByTestId('5');
    expect(fullname).toBeInTheDocument();
    expect(fullname).toHaveTextContent('John Smith');
    expect(professionNameElem).toBeInTheDocument();
    expect(professionNameElem).toHaveTextContent('JS Developer');
    expect(companyNameElem).toBeInTheDocument();
    expect(companyNameElem).toHaveTextContent('Google');
    expect(courseCountElem).toBeInTheDocument();
    expect(courseCountElem).toHaveTextContent('5');
  });
});
*/
