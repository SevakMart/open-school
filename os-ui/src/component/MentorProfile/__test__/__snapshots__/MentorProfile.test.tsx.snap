// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make unit tests on Mentor Profile Make a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <div
      class="headerContainer"
    >
      <div
        class="MailLinkedinIconsContainer"
      >
        <p>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </p>
        <p>
          |
        </p>
        <p>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
            />
          </svg>
        </p>
      </div>
      <p>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(94, 97, 123); font-size: 1rem; cursor: pointer; fill: transparent;"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21l-7-5-7 5V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2z"
          />
        </svg>
      </p>
    </div>
    <div
      class="bodyContainer"
    >
      <div
        class="mentorInfoContainer"
      >
        <div
          class="mentorAvatar"
        >
          <img
            alt="mentor_avatar"
            src="https://image.shutterstock.com/image-vector/businessman-avatar-profile-picture-260nw-221565274.jpg"
          />
        </div>
        <div
          class="mentorInfo"
        >
          <h3
            data-testid="John Smith"
          >
            John Smith
          </h3>
          <p
            data-testid="JS Developer"
          >
            JS Developer
          </p>
          <p
            data-testid="Google"
          >
            Google
          </p>
        </div>
      </div>
      <div
        class="mentorExtraInfo"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 3h6a4 4 0 0 1 4 4v14a3 3 0 0 0-3-3H2z"
          />
          <path
            d="M22 3h-6a4 4 0 0 0-4 4v14a3 3 0 0 1 3-3h7z"
          />
        </svg>
        <p
          data-testid="5"
        >
          5
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
